cmd/nirvana/api/api.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2018 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/caicloud/nirvana"
+	"github.com/caicloud/nirvana/cmd/nirvana/utils"
+	"github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/log"
+	"github.com/caicloud/nirvana/service"
+	"github.com/caicloud/nirvana/utils/generators/swagger"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const mimeHTML = "text/html"
+
+func init() {
+	err := service.RegisterProducer(service.NewSimpleSerializer("text/html"))
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func newAPICommand() *cobra.Command {
+	options := &apiOptions{}
```

Comment:
flags.StringVar(&o.Serve, "serve", o.Server, "Start a server to host api docs")
flags.StringVar(&o.Output, "output", o.Output, "Directory to output api specifications")

add default value to &apiOptions{},  It's easier to add unit tests